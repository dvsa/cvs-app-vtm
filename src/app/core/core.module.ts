import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { GlobalErrorComponent } from './components/global-error/global-error.component';
import { SpinnerComponent } from './components/spinner/spinner.component';
import { FooterComponent } from './components/footer/footer.component';
import { HeaderComponent } from './components/header/header.component';
import { RouterModule } from '@angular/router';
import { BreadcrumbsComponent } from './components/breadcrumbs/breadcrumbs.component';
import { PageNotFoundComponent } from './components/page-not-found/page-not-found.component';
import { ServerErrorComponent } from './components/server-error/server-error.component';

@NgModule({
  declarations: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent, PageNotFoundComponent, ServerErrorComponent],
  imports: [CommonModule, RouterModule],
  exports: [FooterComponent, HeaderComponent, GlobalErrorComponent, SpinnerComponent, BreadcrumbsComponent]
})
export class CoreModule {}
