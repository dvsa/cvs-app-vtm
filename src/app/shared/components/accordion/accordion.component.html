<div class="govuk-accordion__section" [ngClass]="{ 'govuk-accordion__section--expanded': isExpanded }">
  <div class="govuk-accordion__section-header">
    <h2 class="govuk-accordion__section-heading">
      <button
        id="accordion-control-{{ id }}"
        type="button"
        attr.aria-controls="accordion-details-{{ id }}"
        class="govuk-accordion__section-button"
        [attr.aria-expanded]="isExpanded"
        (click)="isExpanded ? close(id) : open(id)"
      >
        <span class="govuk-accordion__section-heading-text" id="accordion-title-{{ id }}">
          <span class="govuk-accordion__section-heading-text-focus">{{ title }}</span>
        </span>
        <span class="govuk-visually-hidden govuk-accordion__section-heading-divider">, </span>
        <span class="govuk-accordion__section-toggle" data-nosnippet="">
          <span class="govuk-accordion__section-toggle-focus">
            <span [class]="iconStyle"></span>
            <span class="govuk-accordion__section-toggle-text">
              {{ isExpanded ? 'Hide' : 'Show' }}<span class="govuk-visually-hidden"> this section</span>
            </span>
          </span>
        </span>
      </button>
    </h2>
  </div>
  <div id="accordion-details-{{ id }}" class="govuk-accordion__section-content" attr.aria-labelledby="accordion-title-{{ id }}">
    <ng-content></ng-content>
  </div>
</div>
