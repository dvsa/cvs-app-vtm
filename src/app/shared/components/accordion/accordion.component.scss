.govuk-accordion__section--expanded {
  .govuk-accordion__section-content {
    display: block;
  }
}
