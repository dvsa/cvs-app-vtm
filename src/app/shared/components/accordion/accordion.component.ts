import {
  ChangeDetectionStrategy, ChangeDetectorRef, Component, Input,
} from '@angular/core';
import { Store } from '@ngrx/store';
import {
  addSectionState, removeSectionState,
} from '@store/technical-records';

@Component({
  selector: 'app-accordion[id]',
  templateUrl: './accordion.component.html',
  styleUrls: ['./accordion.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class AccordionComponent {
  @Input() title: string | undefined = '';
  @Input() id: string | number = '';

  @Input() isExpanded: boolean | null | undefined = false;

  constructor(private cdr: ChangeDetectorRef, private store: Store) {}

  get iconStyle(): string {
    return `govuk-accordion-nav__chevron${(this.isExpanded ? '' : ' govuk-accordion-nav__chevron--down')}`;
  }

  open(sectionName: string | number | undefined): void {
    this.isExpanded = true;
    this.cdr.markForCheck();
    if (sectionName) this.store.dispatch(addSectionState({ section: sectionName }));
  }

  close(sectionName: string | number | undefined): void {
    this.isExpanded = false;
    this.cdr.markForCheck();
    if (sectionName) this.store.dispatch(removeSectionState({ section: sectionName }));
  }
}
