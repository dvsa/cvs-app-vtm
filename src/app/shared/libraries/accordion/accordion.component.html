<div class="govuk-accordion" data-module="govuk-accordion" id="accordion-default">
  <ng-content></ng-content>
</div>



