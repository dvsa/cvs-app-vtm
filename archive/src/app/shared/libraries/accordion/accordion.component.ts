import { Component, OnInit, ChangeDetectionStrategy } from '@angular/core';

@Component({
  selector: 'vtm-accordion',
  templateUrl: './accordion.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AccordionComponent implements OnInit {
  constructor() {}

  ngOnInit() {}
}
