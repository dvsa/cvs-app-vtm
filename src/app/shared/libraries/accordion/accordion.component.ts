import {Component, OnInit, ChangeDetectionStrategy} from '@angular/core';

@Component({
  selector: 'vtm-accordion',
  templateUrl: './accordion.component.html',
  styleUrls: ['./accordion.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AccordionComponent implements OnInit {

  constructor() {
  }

  ngOnInit() {
  }

}
