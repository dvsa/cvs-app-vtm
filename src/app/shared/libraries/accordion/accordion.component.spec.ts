import { async, ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';

import { AccordionComponent } from './accordion.component';
import { Component, DebugElement } from '@angular/core';
import { By } from '@angular/platform-browser';

describe('AccordionComponent', () => {
  let component: AccordionComponent;
  let fixture: ComponentFixture<AccordionComponent>;

  let testComponent: TestAccordionComponent;
  let testFixture: ComponentFixture<TestAccordionComponent>;
  let element: DebugElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [AccordionComponent, TestAccordionComponent]
    }).compileComponents();
  }));

  beforeEach(fakeAsync(() => {
    fixture = TestBed.createComponent(AccordionComponent);
    component = fixture.componentInstance;
    testFixture = TestBed.createComponent(TestAccordionComponent);
    testComponent = testFixture.componentInstance;
    testFixture.detectChanges();
    tick();
    element = testFixture.debugElement.query(By.css('[name=testAccordion]'));
    fixture.detectChanges();
  }));

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should populate the accordion with provided content', fakeAsync(() => {
    testFixture.detectChanges();
    tick();

    expect(element.nativeElement.innerHTML).toEqual('Test');
    expect(testFixture).toMatchSnapshot();
  }));
});

@Component({
  selector: 'test-accordion',
  template: `
    <p name="testAccordion">Test</p>
  `
})
class TestAccordionComponent {}
