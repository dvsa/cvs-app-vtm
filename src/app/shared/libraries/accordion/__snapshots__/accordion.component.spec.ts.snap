// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionComponent should create 1`] = `
<vtm-accordion>
  <div
    class="govuk-accordion"
    data-module="govuk-accordion"
    id="accordion-default"
  />
</vtm-accordion>
`;

exports[`AccordionComponent should populate the accordion with provided content 1`] = `
<test-accordion>
  <p
    name="testAccordion"
  >
    Test
  </p>
</test-accordion>
`;
